core/common/src/files/FileSystem.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io.files
+
+import kotlinx.io.IOException
+import kotlinx.io.Sink
+import kotlinx.io.Source
+
+/**
+ * An interface providing basic operations on a file system.
+ *
+ * **This API is unstable and subject to change.**
+ */
+public sealed interface FileSystem {
+    /**
+     * Path to a directory suitable for creating temporary files.
+     */
+    public val temporaryDirectory: Path
+
+    /**
+     * Returns `true` if there is a file system entity corresponding to a [path],
+     * otherwise returns `false`.
+     *
+     * @param path the path that should be checked for existence.
+     *
+     * @throws kotlinx.io.IOException when the attempt to check the existence of the [path] failed.
+     */
+    public fun exists(path: Path): Boolean
+
+    /**
+     * Deletes [path] from a file system. If there is no file system entity
```

Comment:
The doc is a bit confusing from the user's perspective.
This function deletes a file by the given path (and/or directory?), not the path itself

Reply:
Fixed